src/strgen/strgen_base.cpp

```diff
@@ -967,16 +964,25 @@ void LanguageWriter::WriteLang(const StringData &data)
 					buffer.AppendByte(0);
 					/* Write string */
 					PutCommandString(&buffer, c.string.c_str());
-					buffer.AppendByte(0); // terminate with a zero
 					/* Fill in the length */
 					uint size = (uint)buffer.size() - (pos + 2);
-					buffer[pos + 0] = GB(size, 8, 8);
-					buffer[pos + 1] = GB(size, 0, 8);
+					buffer[pos + 0] = GB(size, 0, 8);
+					buffer[pos + 1] = GB(size, 8, 8);
 				}
+
+				default_case_pos = (uint)buffer.size();
```

Comment:
```suggestion
				default_case_pos = static_cast<uint>(buffer.size());
```

Reply:
Strgen generally needs some love. There are tons of weird casts and types.
I did not want to make it inconsistent by using static_cast in some places, and leaving C casts in other places.